crates/sui-core/src/authority/authority_store.rs

```diff
@@ -340,14 +392,32 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
     pub fn get_missing_input_objects(
         &self,
         digest: &TransactionDigest,
+        valid_effects: Option<ValidTransactionEffects>,
         objects: &[InputObjectKind],
     ) -> Result<Vec<ObjectKey>, SuiError> {
         let shared_locks_cell: OnceCell<HashMap<_, _>> = OnceCell::new();
 
+        let shared_input_versions: Option<HashMap<ObjectID, SequenceNumber>> =
```

Comment:
`handle_certificate_with_effects()` is called for execution later, which sets shared object locks from the effect. Similar to the case of handling shared object transactions with effects, it seems more consistent and simpler to set the shared object locks from effects before calling `TransactionManager::enqueue()`. Then `get_missing_input_objects()` can work as it is. Otherwise it seems inconsistent that we use two ways to get shared object lock versions with effects depending on the stage of processing. What do you think?

Reply:
that's a good point, I think we can do it that way.